docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
+* Keeping the Cluster object up to date with external objects.
+* Creating a kubeconfig secret for child clusters.
+
+## Contracts
+
+### Infrastructure Provider
+
+The InfrastructureCluster object **must** have a status field.
+
+In the status field it must have:
+
+* `ready` - a boolean field
+* `apiEndpoints` - a slice of strings
+
+Example:
+```yaml
+kind: MyCluster
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+status:
+    ready: true
+    apiEndpoints:
+    - example.com:3333
+    - example.com:3334
+```
+
+### Secrets
+
+| secret name | field name | content |
+|:---:|:---:|:---:|
+|`<cluster-name>-ca`|`tls.crt`|base64 encoded TLS certificate in PEM format|
+|`<cluster-name>-ca`|`tls.key`|base64 encoded TLS private key in PEM format|
+
+OR
+
+| secret name | field name | content |
+|:---:|:---:|:---:|
+|`<cluster-name>-kubeconfig`|none required|none required|
```

Comment:
The filed name should be `value` and the content should be a `base64 encoded kubeconfig`

Reply:
Do you think it's worth adding that check to `ClusterReconciler.reconcileKubeconfig`Or do you think it's enough to look for a properly named secret? Right now that function is only checking for the existence of the secret.

But yes, this is the expected value, I'll definitely add these changes, thanks

Reply:
It should definitely error today if the value isn't found, I can dig up the code later

Reply:
This only checks for the existence of the secret, not the contents.

https://github.com/kubernetes-sigs/cluster-api/blob/2c4cbfa482d3bc7857333aeb1a0eb0599e2320bd/controllers/cluster_controller_phases.go#L181-L188